pkg/sfu/sendsidebwe/packet_tracker.go

```diff
@@ -0,0 +1,73 @@
+package sendsidebwe
+
+import (
+	"sync"
+	"time"
+
+	"github.com/livekit/protocol/logger"
+)
+
+type packetInfo struct {
+	sendTime    time.Time
+	arrivalTime int64
+	headerSize  uint16
+	payloadSize uint16
+}
+
+type PacketTracker struct {
+	logger logger.Logger
+
+	lock        sync.RWMutex
+	initialized bool
+	highestSN   uint16
+	packetInfos [1 << 16]packetInfo
```

Comment:
I am justing using 64K array size to cover the full sequence number space. At 20 bytes per entry, that is 1.28 MB. Went with this to simplify as that is per participant. So, for a 1000 participant server, it will consume 1.28 GB which is not great, but not too bad either. Will look at optimising this later. Will make the code more complicated.